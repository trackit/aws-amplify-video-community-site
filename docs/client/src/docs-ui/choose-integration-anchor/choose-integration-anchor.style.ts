import {css} from "emotion";

export const hostStyle = css`
  display: block;
  padding-bottom: 2rem;

  h3,
  h4 {
    display: block;
  }

  h3 {
    border-bottom: 0.125rem solid var(--border-color);
    padding: 1rem 0;
    margin-bottom: 1rem;
  }
`;
