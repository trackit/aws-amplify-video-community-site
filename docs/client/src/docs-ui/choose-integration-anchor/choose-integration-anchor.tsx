import {Component, Host, h, Prop} from "@stencil/core";
import {Page} from "../../api";
import {
  mobileFilterMetadataByOption,
  webFilterMetadataByOption,
} from "../../utils/filter-data";
import {hostStyle} from "./choose-integration-anchor.style";

@Component({tag: "docs-choose-integration-anchor", shadow: false})
export class DocsChooseIntegrationAnchor {
  /*** the current page's data */
  @Prop() readonly page?: Page;

  render() {
    return (
      <Host class={hostStyle}>
        <h2>Choose a platform or framework</h2>
        <h3>Web</h3>
        <amplify-responsive-grid
          gridGap={1}
          columns={5}
          class="border-radius margin-top-md"
        >
          {Object.entries(webFilterMetadataByOption).map(
            ([filterValue, {label, graphicURI}]) => {
              const route =
                this.page && `${this.page.route}/q/integration/${filterValue}`;

              return (
                <docs-card
                  key={label}
                  url={route}
                  QSPs={`?sc_icampaign=${filterValue}-start&sc_ichannel=choose-integration`}
                >
                  <img slot="graphic" src={graphicURI} alt={`${label} Logo`} />
                  <h4 slot="heading">{label}</h4>
                </docs-card>
              );
            },
          )}
        </amplify-responsive-grid>
        <h3>Mobile</h3>
        <amplify-responsive-grid
          gridGap={1}
          columns={5}
          class="border-radius margin-top-md"
        >
          {Object.entries(mobileFilterMetadataByOption).map(
            ([filterValue, {label, graphicURI}]) => {
              const route =
                this.page && `${this.page.route}/q/integration/${filterValue}`;

              return (
                <docs-card
                  key={label}
                  url={route}
                  QSPs={`?sc_icampaign=${filterValue}-start&sc_ichannel=choose-integration`}
                >
                  <img slot="graphic" src={graphicURI} alt={`${label} Logo`} />
                  <h4 slot="heading">{label}</h4>
                </docs-card>
              );
            },
          )}
        </amplify-responsive-grid>
      </Host>
    );
  }
}
