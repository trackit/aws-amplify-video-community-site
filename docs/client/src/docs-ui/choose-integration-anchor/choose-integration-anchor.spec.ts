import {DocsChooseIntegrationAnchor} from "./choose-integration-anchor";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-choose-integration-anchor", () => {
  describe("Component logic", () => {
    let chooseAnchor: DocsChooseIntegrationAnchor;
    beforeEach(() => (chooseAnchor = new DocsChooseIntegrationAnchor()));

    it("should init `page` as `undefined`", () =>
      expect(chooseAnchor.page).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsChooseIntegrationAnchor],
            html: `<docs-choose-integration-anchor />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
