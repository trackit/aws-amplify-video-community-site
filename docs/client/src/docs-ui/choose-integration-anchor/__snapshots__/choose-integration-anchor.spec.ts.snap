// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-choose-integration-anchor Render logic should render 1`] = `
<docs-choose-integration-anchor class="css-19gh8xg">
  <h2>
    Choose a platform or framework
  </h2>
  <h3>
    Web
  </h3>
  <amplify-responsive-grid class="border-radius margin-top-md" columns="5" gridgap="1">
    <docs-card qsps="?sc_icampaign=js-start&amp;sc_ichannel=choose-integration">
      <img alt="JavaScript Logo" slot="graphic" src="/assets/integrations/js.svg">
      <h4 slot="heading">
        JavaScript
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=react-start&amp;sc_ichannel=choose-integration">
      <img alt="React Logo" slot="graphic" src="/assets/integrations/react.svg">
      <h4 slot="heading">
        React
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=angular-start&amp;sc_ichannel=choose-integration">
      <img alt="Angular Logo" slot="graphic" src="/assets/integrations/angular.svg">
      <h4 slot="heading">
        Angular
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=vue-start&amp;sc_ichannel=choose-integration">
      <img alt="Vue Logo" slot="graphic" src="/assets/integrations/vue.svg">
      <h4 slot="heading">
        Vue
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=next-start&amp;sc_ichannel=choose-integration">
      <img alt="Next.js Logo" slot="graphic" src="/assets/integrations/next.svg">
      <h4 slot="heading">
        Next.js
      </h4>
    </docs-card>
  </amplify-responsive-grid>
  <h3>
    Mobile
  </h3>
  <amplify-responsive-grid class="border-radius margin-top-md" columns="5" gridgap="1">
    <docs-card qsps="?sc_icampaign=android-start&amp;sc_ichannel=choose-integration">
      <img alt="Android Logo" slot="graphic" src="/assets/integrations/android.svg">
      <h4 slot="heading">
        Android
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=ios-start&amp;sc_ichannel=choose-integration">
      <img alt="iOS Logo" slot="graphic" src="/assets/integrations/ios.svg">
      <h4 slot="heading">
        iOS
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=react-native-start&amp;sc_ichannel=choose-integration">
      <img alt="React Native Logo" slot="graphic" src="/assets/integrations/react-native.svg">
      <h4 slot="heading">
        React Native
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=ionic-start&amp;sc_ichannel=choose-integration">
      <img alt="Ionic Logo" slot="graphic" src="/assets/integrations/ionic.svg">
      <h4 slot="heading">
        Ionic
      </h4>
    </docs-card>
    <docs-card qsps="?sc_icampaign=flutter-start&amp;sc_ichannel=choose-integration">
      <img alt="Flutter Logo" slot="graphic" src="/assets/integrations/flutter.svg">
      <h4 slot="heading">
        Flutter
      </h4>
    </docs-card>
  </amplify-responsive-grid>
</docs-choose-integration-anchor>
`;
